### ISSUE TITLE
[feature request] Quasi recurrent neural networks (QRNN) in tensorflow 

### ISSUE THREAD

**Issue by dhwajraj** [tensorflow/tensorflow#19965]
It would be cool to have native Quasi recurrent neural networks (QRNN) implementation included in the tensorflow package.

Following is the paper : Bradbury, James, et al. "Quasi-recurrent neural networks." arXiv preprint arXiv:1611.01576 (2016). [link](https://arxiv.org/pdf/1611.01576)
It is claimed in the paper that QRNN is 14-16x faster than cudnn_lstm while giving same accuracies.

reference : https://github.com/salesforce/pytorch-qrnn

**System information**
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
TensorFlow installed from (source or binary): binary
TensorFlow version (use command below): 1.8.0
Python version: 3.6.5
Bazel version (if compiling from source): none
GCC/Compiler version (if compiling from source): none
CUDA/cuDNN version: none
GPU model and memory: none
Exact command to reproduce: none

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by zhbbupt** [comment]
I agree with that.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 49 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 80 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 95 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 110 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @sguada: It has been 125 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by timothyslau** [comment]
If it helps @ebrevdo, someone has tried implementing it in Tensorflow already:
https://github.com/JonathanRaiman/tensorflow_qrnn

**Comment by ebrevdo** [comment]
Nice!  Yeah; @JonathanRaiman has done some really great stuff.  At this point, I think getting QRNN into Keras (as that's where it would have to land in TF2) would be an uphill climb; but I'll let @martinwicke comment.  I know QRNNs are quite popular so it may be possible.  If not, then I'm sure it would be a good fit for the TF-Addons repo.

**Comment by tilakrayal** [comment]
@dhwajraj,
Could you please take a look at this comment from the [developer](https://github.com/tensorflow/tensorflow/issues/19965#issuecomment-639978167) and let us know if the issue still persists. Thank you!
